src/main/java/org/apache/maven/plugins/scripting/AbstractScriptEvaluator.java

```diff
@@ -32,13 +35,17 @@ abstract class AbstractScriptEvaluator {
 
     /**
      * @param bindings not null bindings to provide to the script to execute
+     * @param log engine logger if context aware.
      * @return the possibly null result the script produced
      * @throws UnsupportedScriptEngineException when the engine is not configured correctly
      * @throws ScriptException  if an error occurs in script.
      */
-    public final Object eval(Bindings bindings) throws ScriptException, UnsupportedScriptEngineException {
+    public final Object eval(Bindings bindings, Log log) throws ScriptException, UnsupportedScriptEngineException {
         ScriptEngineManager manager = new ScriptEngineManager();
         ScriptEngine engine = getEngine(manager);
+        if (ContextAwareEngine.class.isInstance(engine)) {
```

Comment:
This feels somewhat inside out to me. I really wouldn't expect the log to be injected here. It feels like the engine should already know its log.

Reply:
```suggestion
        if (engine instanceof ContextAwareEngine) {
```

That would be sufficient to me. Yes, it should have already known its log, but THAT would be a public API change.